microceph/api/remote.go

```diff
@@ -0,0 +1,147 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/canonical/lxd/lxd/response"
+	"github.com/canonical/lxd/shared/logger"
+	"github.com/canonical/microceph/microceph/api/types"
+	"github.com/canonical/microceph/microceph/ceph"
+	"github.com/canonical/microceph/microceph/client"
+	"github.com/canonical/microceph/microceph/constants"
+	"github.com/canonical/microceph/microceph/database"
+	"github.com/canonical/microceph/microceph/interfaces"
+	"github.com/canonical/microcluster/rest"
+	"github.com/canonical/microcluster/state"
+	"github.com/gorilla/mux"
+)
+
+var remoteCmd = rest.Endpoint{
+	Path: "client/remotes",
+	Get:  rest.EndpointAction{Handler: CmdRemoteGet, ProxyTarget: false},
+}
+
+var remoteNameCmd = rest.Endpoint{
+	Path:   "client/remotes/{name}",
+	Put:    rest.EndpointAction{Handler: CmdRemotePut, ProxyTarget: false},
+	Get:    rest.EndpointAction{Handler: CmdRemoteGet, ProxyTarget: false},
+	Delete: rest.EndpointAction{Handler: CmdRemoteDelete, ProxyTarget: false},
+}
+
+var CmdRemotePut = func(state *state.State, r *http.Request) response.Response {
+	var req types.Remote
+
+	err := json.NewDecoder(r.Body).Decode(&req)
+	if err != nil {
+		return response.InternalError(err)
+	}
+
+	err = RenderConfAndKeyringFiles(req.Name, req.LocalName, req.Config)
+	if err != nil {
+		return response.InternalError(fmt.Errorf("couldn't render files: %w", err))
+	}
+
+	if !req.RenderOnly {
+		// Asynchronously persist this on db and send request to other cluster members.
+		go func() {
```

Comment:
I'm not sure if we should return before this call completes. Why not make this a synchronous call?